udev-trigger-generator

```diff
@@ -0,0 +1,27 @@
+#! /bin/bash
+# Copyright (c) 2020 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# bsc#1174287: On PowerPC, exclude memory devices from udev coldplug
+export PATH=/usr/bin
+# /tmp may not be writable yet
+export TMPDIR=/dev/shm
+trap 'echo $0: error in "$BASH_COMMAND" >&2' ERR
+
+[[ $(uname -m) = ppc64le ]] || exit 0
+case $(systemd-detect-virt) in
+    none|powervm) ;;
+    *) exit 0;;
+esac
+
+DIR=$1
+[[ $DIR ]]
```

Comment:
what is it for ?

Reply:
I don't think it tests for the presence of a directory. `[[ $DIR ]]` only fails if $DIR is the empty string, which can't happen.

Reply:
You're right, sorry. Still warming up after vacation ... I'll change this to 

    [[ $DIR  && -d $DIR ]]